crates/matrix-sdk/src/notification_settings/rules.rs

```diff
@@ -0,0 +1,1150 @@
+                            x,
+                            PushCondition::EventMatch { key, pattern } if key == "room_id" && *pattern == *room_id
+                        )))
+                    }
+                    _ => {
+                        panic!("A NewPushRule::Override is expected.")
+                    }
+                }
+            }
+            _ => {
+                panic!("A Command::SetPushRule is expected.")
+            }
+        }
+    }
+
+    #[async_test]
+    async fn insert_room_rule_room() {
+        let room_id = RoomId::parse("!AAAaAAAAAaaAAaaaaa:matrix.org").unwrap();
+        let user_id = UserId::parse("@user:matrix.org").unwrap();
+        let ruleset = Ruleset::server_default(&user_id);
+        let mut rules = Rules::new(ruleset);
+
+        let commands = rules.insert_room_rule(RuleKind::Room, &room_id, true).unwrap();
+
+        // The ruleset should contains the new rule
+        let new_rule =
+            rules.ruleset.get(RuleKind::Room, &room_id).expect("a new Room rule is expected.");
+        match new_rule.to_owned() {
+            AnyPushRule::Room(rule) => {
+                assert_eq!(rule.rule_id, room_id);
+            }
+            _ => {
+                panic!("An AnyPushRule::Room is expected.")
+            }
+        }
+
+        // The command list should contains only a SetPushRule command
+        assert_eq!(commands.len(), 1);
+        match &commands[0] {
+            Command::SetPushRule(scope, rule) => {
+                assert_eq!(scope, &RuleScope::Global);
+                match rule {
+                    NewPushRule::Room(rule) => {
+                        assert_eq!(rule.rule_id, room_id);
+                    }
+                    _ => {
+                        panic!("A NewPushRule::Room is expected.")
+                    }
+                }
+            }
+            _ => {
+                panic!("A Command::SetPushRule is expected.")
+            }
+        }
+    }
+
+    #[async_test]
+    async fn insert_room_rule_invalid_kind() {
+        let room_id = RoomId::parse("!AAAaAAAAAaaAAaaaaa:matrix.org").unwrap();
+        let user_id = UserId::parse("@user:matrix.org").unwrap();
+        let ruleset = Ruleset::server_default(&user_id);
+        let mut rules = Rules::new(ruleset);
+
+        rules
+            .insert_room_rule(RuleKind::Content, &room_id, true)
+            .expect_err("An InvalidParameter error is expected");
+    }
+
+    #[async_test]
+    async fn delete_rules() {
+        let room_id = RoomId::parse("!AAAaAAAAAaaAAaaaaa:matrix.org").unwrap();
+        let user_id = UserId::parse("@user:matrix.org").unwrap();
+        let ruleset = Ruleset::server_default(&user_id);
+        let mut rules = Rules::new(ruleset);
+
+        rules
+            .delete_rules(vec![(RuleKind::Room, room_id.to_string())], vec![])
+            .expect_err("An error is expected while trying to delete an unknown rule.");
```

Comment:
Could we match on the specific error we're expecting, here?

Reply:
The only possible error here is `RemovePushRuleError`, so I don't know how to enhance this one.